### PR TITLE
fixes 'add docker-for-desktop platform to kfctl (golang)'

### LINKED ISSUE #2420: add docker-for-desktop platform to kfctl (golang)
this platform is very (if not identical to minikube)

### PR DESCRIPTION
**Pull request by kkasravi**
fixes #2420 

this PR is based off of PR #2428

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2433)

### CODE REVIEW

**Comment by jlewi** [comment]
Marking this as WIP until #2428 is merged.

File: bootstrap/Makefile [hunk 1]
```diff
@@ -80,26 +90,41 @@ run-local-docker:
...
+	PLUGINS_ENVIRONMENT=$(PLUGINS_ENVIRONMENT) kfctl init $(HOME)/minikube -V --platform minikube --version master && \
+	rm -rf $(HOME)/docker-for-desktop && \
+	PLUGINS_ENVIRONMENT=$(PLUGINS_ENVIRONMENT) kfctl init $(HOME)/docker-for-desktop -V --platform docker-for-desktop --version master && \
 	echo SUCCESS
 
 # generate all --email john@foo.com --ipName 35.233.240.120
 test-known-platforms-generate: test-known-platforms-init
 	@cd ~/ksonnet && \
-	kfctl generate all -V && \
+	PLUGINS_ENVIRONMENT=$(GOPATH)/src/github.com/kubeflow/kubeflow/bootstrap/bin kfctl generate all -V && \
```
Comment (ashahba) [inline bootstrap/Makefile]: I'd just do:
```
PLUGINS_ENVIRONMENT=$(PLUGINS_ENVIRONMENT)
```

File: bootstrap/Makefile [hunk 2]
```diff
@@ -80,26 +90,41 @@ run-local-docker:
...
 	cd ~/minikube && \
-	kfctl generate all -V --mount-local && \
+	PLUGINS_ENVIRONMENT=$(GOPATH)/src/github.com/kubeflow/kubeflow/bootstrap/bin kfctl generate all -V --mount-local && \
```
Comment (ashahba) [inline bootstrap/Makefile]: same as above

File: bootstrap/Makefile [hunk 3]
```diff
@@ -80,26 +90,41 @@ run-local-docker:
...
+	cd ~/docker-for-desktop && \
+	PLUGINS_ENVIRONMENT=$(GOPATH)/src/github.com/kubeflow/kubeflow/bootstrap/bin kfctl generate all -V --mount-local && \
```
Comment (ashahba) [inline bootstrap/Makefile]: same as above

File: bootstrap/Makefile [hunk 4]
```diff
@@ -80,26 +90,41 @@ run-local-docker:
...
 	cd ~/foo && \
 	PLUGINS_ENVIRONMENT=$(GOPATH)/src/github.com/kubeflow/kubeflow/bootstrap/bin kfctl generate all -V && \
```
Comment (ashahba) [inline bootstrap/Makefile]: same as above

File: bootstrap/cmd/kfctl/README.md [hunk 5]
```diff
@@ -222,28 +222,16 @@ make test-known-platforms-generate
...
-		}
-		return symbol.(func(map[string]interface{}) kftypes.KfApp)(options), nil
-*/
-		return nil, fmt.Errorf("unknown platform %v", platform)
-	}
+```
+make debug
+```
+
+To go back to where plugins are loaded (this should be the default) run
```
Comment (ashahba) [inline bootstrap/cmd/kfctl/README.md]: Not sure about the wording but a better message would be a good idea

Comment (kkasravi) [reply bootstrap/cmd/kfctl/README.md]: thanks Ebi - we're going to move most of the plugins back to a static configuration to minimize  complexity. The PLUGINS_ENVIRONMENT will probably be set to a specific directory when we add the brew installer.

**Comment by kkasravi** [comment]
@jlewi @ashahba this is ready for review

**Comment by jlewi** [comment]
just a thought

If you wanted a platform to experiment with the plugin model. docker-for-desktop might be a good one.
It looks like we don't have explicit instructions for it on the website
https://www.kubeflow.org/docs/


/cc @rogaha who added docker for desktop scripts in #1954

**Comment by kkasravi** [comment]
@jlewi replaced foo with dockerfordesktop

**Pull request merged**